pkg/providers/vsphere/session/session_vm_update.go

```diff
@@ -1099,3 +1104,44 @@ func (s *Session) UpdateVirtualMachine(
 
 	return err
 }
+
+func isVMPaused(
+	vmCtx context.VirtualMachineContext,
+	moVM *mo.VirtualMachine) bool {
+
+	vm := vmCtx.VM
+	var adminPaused, devopsPaused bool
+
+	for i := range moVM.Config.ExtraConfig {
```

Comment:
can reuse isPausedByAdmin() here too?. and move isPausedByAdmin to a util directory